bep028spec.md

```diff
@@ -428,382 +369,679 @@ Environment records are OPTIONAL. If included, each environment record MUST have
+```
+└─ example_dataset
+   ├─ prov/
+   │  └─ prov-dcm2niix_base.json
+   ├─ sub-001/
+   │  └─ ses-01/
+   │     └─ anat/
+   │        ├─ sub-001_ses-01_T1w.nii.gz
+   │        └─ sub-001_ses-01_T1w.json
+   ├─ sub-002/
+   │  └─ ses-01/
+   │     └─ anat/
+   │        ├─ sub-002_ses-01_T1w.nii.gz
+   │        └─ sub-002_ses-01_T1w.json
+   ├─ ...
+   └─ dataset_description.json
+```
 
+Inside the sidecar JSON, the `GenearatedBy` field must describe the `Activity` that generated the data file, either with a reference to an existing `Id`:
 
-                	"bids_dir" : "data/bids_root",
+```JSON
+{
+    "GeneratedBy": "urn:conversion-00f3a18f",
+}
+```
 
+or with a complete description of the `Activity` if it was not described elsewhere.
 
-                	"output_dir" : "out/",
+```JSON
+{
+    "GeneratedBy": {
+        "Id": "urn:conversion-00f3a18f",
+        "Label": "Conversion",
+        "Command": "convert -i raw_file.ext -o sub-001_ses-01_T1w.nii.gz"
+    }
+}
+```
 
+Based on the same principle, the `SidecarGenearatedBy` field can be defined to describe the `Activity` that generated the sidecar JSON file.
+If the `SidecarGenearatedBy` field is not defined, BIDS-Prov assumes that the sidecar JSON was generated by the `Activity` described in the `GenearatedBy` field.
 
-                	"anaysis_level" : "participant"
+No other field is allowed to describe provenance inside sidecar JSONs.
 
-    		},
+> [!WARNING]
+> When using sidecar JSON files to describe provenance, the `@context` and `BIDSProvVersion` fields MUST be defined inside a `*_base.json` file, e.g.:
+> ```JSON
+> {
+>     "@context": "https://purl.org/nidash/bidsprov/context.json",
+>     "BIDSProvVersion": "0.0.1"
+> }
 
+#### 3.2.2 Subdirectories level provenance
 
-        	"used": "niiri:bids_data1"
+BIDS-Prov files can be stored in a `prov/` directory in any subdirectory of the dataset (or BIDS-Derivatives directories).
 
-  	},
+In this case, the provenance metadata applies to the data files inside or below in the directory tree ; as stated by [BIDS common principles](https://bids-specification.readthedocs.io/en/stable/common-principles.html#filesystem-structure).
 
-	],
+Each BIDS-Prov file MUST meet the following naming convention. The `label` of the `prov` entity is arbitrary, `suffix` is one of listed in [3.3.1 Suffixes](#3-1-3-suffixes), and `extension` is either `json` or `jsonld`.
 
-	"prov:Entity": [
+```
+sub-<label>/
+  [ses-<label>/]
+    prov/
+      sub-<label>[_ses-<label>]_prov-<label>_<suffix>.<extension>
+```
 
+> [!TIP]
+> Here is an example dataset tree:
+> ```
+> └─ dataset
+>    ├─ prov/
+>    │  └─ prov-dcm2niix_base.json
+>    ├─ sub-001/
+>    │  ├─ prov/
+>    │  │  └─ sub-001_prov-dcm2niix_act.json
+>    │  └─ ses-01/
+>    │     ├─ prov/
+>    │     │  └─ sub-001_ses-01_prov-dcm2niix_act.json
+>    │     └─ ...
+>    ├─ sub-002/
+>    │  ├─ prov/
+>    │  │  └─ sub-002_prov-dcm2niix_act.json
+>    │  └─ ...
```

Comment:
i'll comment here. can you point to the discussion about folder level messages? i'm not saying we shouldn't do this, just want to see the discussion.

Reply:
Hi @satra - this is a proposal following up discussions with @yarikoptic in #150 #148 #144 #146 (and more). If you are up for a call maybe we can review this together so that you can let us know your thoughts (see email w/ proposed times/dates)? This PR is intended as a basis for those discussions. Let us know!